### PR TITLE
Add KNN classification to PILatentEvalVideo

### PR DESCRIPTION
**Pull request by ElmoPA**
5-fold CV KNN classifier per layer slice predicting embodiment from key vectors. Uses cuML GPU when available, sklearn CPU fallback. New evaluator knobs compute_knn and knn_k. Logs accuracy at INFO so base vs finetuned diff surfaces in the run log directly.

### CODE REVIEW

**Comment by ElmoPA** [comment]
* **#486** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/486?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#485** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/485?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#484** <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/484?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/GaTech-RL2/EgoVerse/484?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
